### DISCUSSION TITLE
Is there a easy way to override pydantic error message in FastApi?

### DISCUSSION THREAD

**Discussion by binaryantcom** [fastapi/fastapi#9774] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, Body, HTTPException
from pydantic import BaseModel, Field, validator
import uvicorn

app = FastAPI()

class User(BaseModel):
    name: str = Field(..., alias="username", error_messages={
            "required": "username is required",
            "min_length": "username must be at least 3 characters",
        })
    age: int = Field(..., gt=0, lt=150, alias="user_age", error_messages={
            "required": "user age is required----",
            "gt": "user age must be greater than 0---",
            "lt": "user age must be less than 150---",
        })

@app.post("/users/")
async def create_user(user: User = Body(...)):
    try:
        return {"user": user}
    except ValueError as e:
        raise HTTPException(status_code=400, detail=str(e))


if __name__=='__main__':
    uvicorn.run('test:app',host='localhost',port=5000)
```

### Description

This code is from ChatGPT. This is the style I want, just like C#:[Required(ErrorMessage="My custom error message")]

### Operating System

Windows

### FastAPI Version

0.99

### Python Version

3.10

**Comment by jgould22** [comment, marked as answer]
https://docs.pydantic.dev/1.10/usage/models/#custom-errors for pydantic errors or https://fastapi.tiangolo.com/tutorial/handling-errors/#override-request-validation-exceptions for the errors that fastapi returns when validation fails

**Comment by avaziman** [comment]
I wrote a little exception handler that uses inspect to get the field's description to be used as the error message.
This allows for very easily customizable error messages for each field;

class MySchema(BaseModel):
    name: Annotated[str, Field(description="Name must be between 3 and 100 characters", min_length=3, max_length=100)]
    
   ```py
   # "hacky" way to add field description as error message in 422 erros for easily customizable errors
@app.exception_handler(RequestValidationError)
async def validation_exception_handler(request: Request, exc: RequestValidationError):
    errors = exc.errors()
    custom_errors = []

    # Extract the route's handler function
    route_handler = request.scope["route"].endpoint

    # Get the function's signature and find the schema model
    sig = inspect.signature(route_handler)
    model = None

    for param in sig.parameters.values():
        if hasattr(param.annotation, "model_fields"):  # Check if it's a Pydantic model
            model = param.annotation
            break

    for error in errors:
        loc = ".".join(map(str, error["loc"]))  # Get field path
        field_name = error["loc"][-1]

        # Get field metadata dynamically
        field_info = model.model_fields.get(field_name) if model else None
        custom_message = field_info.description if field_info and field_info.description else error["msg"]
        
        custom_errors.append({
            **error,
            "human_error": custom_message
        })

    return JSONResponse(status_code=422, content={"detail": custom_errors})
    ```
